jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/importer/parser/PayloadParser.java

```diff
@@ -90,4 +105,37 @@ public void writeTo(OutputStream os) throws IOException {
     os.write(payload, 0, payloadLength);
   }
 
+
+
+   public byte[] removeHeader(byte[] packet) {
+      String hexPacket = HexUtils.toHexString(packet);
+      String hexPacketParsed = "";
+      for (String key : msgStartFlags.keySet()) {
+         String startFlag = msgStartFlags.get(key);
+         int startIndex = hexPacket.toLowerCase().indexOf(startFlag);
+         if (hexPacketParsed == ""){
+            logger.debug("Start index for: " + key + " is: " + startIndex);
+            if (startIndex == -1) {
+               logger.debug("Message does not have header for: " + key);
+               break;
+            } else if (startIndex <= HEADER_SIZE) {
+               logger.debug("Message has supported header. startIndex: " + startIndex + " msgFlag: " + startFlag);
+               hexPacketParsed = hexPacket;
+            // Using a value of 35 as the largest index from preliminary testing data.
+            } else if (startIndex > HEADER_SIZE && startIndex < 35) {
```

Comment:
Maybe make 35 consistent with HEADER_SIZE?

Reply:
To add clarity, I will set the 35 value to a constant named: "HEADER_SIZE_WSMP", as this logic is intended to remove the unsupported WSMP payload header from the message.
We hope to support WSMP in the future, but we currently don't have the ability to process the header contents.